### DISCUSSION TITLE
Add support for USB flash drives

### DISCUSSION THREAD

**Discussion by dmarra** [microsoft/WSL#6590] [General]
Currently we are unable to mount USB flash drives. For those of us who may have Bitlocker encrypted flash drives containing files we need to access, this makes it impossible to leverage WSL. 

I would like to be able to see all devices that are mounted in windows mounted to the Linux filesystem as well. 

We should be able to do the following:

1. Access all drives that are accessible within windows, within the subsystem as mounted drives
2. Access Bitlocked drives (if we do not have the ability already)

Likely item 2 needs to be a separate issue, but I wanted to list it here since it is part of my personal use case.

**Comment by DHowett** [comment]
> currently we are unable

Really? What happens when you follow the [first thing i found on Bing for your query](https://www.howtogeek.com/331053/how-to-mount-removable-drives-and-network-locations-in-the-windows-subsystem-for-linux/)?

**Comment by dmarra** [comment]
Look at the limitations when I _literally_ look at the documentation:
https://docs.microsoft.com/en-us/windows/wsl/wsl2-mount-disk.



<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by DHowett** [comment]
I’m talking about using `mount -t drvfs`, which attaches any filesystem that _Windows_ can see. The new `wsl --mount` option attaches physical block storage, which wouldn’t be accessible if BitLockered even if attachable.

**Comment by therealkenc** [comment]
There is [dislocker](https://github.com/Aorimn/dislocker). Can't say I've used it, but people might see if it can be made to work with WSL2.
